pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15167581?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15167581?buildTab=artifacts#/) on master @ [7756496a7d7648c7cc19c83a184deb93817d30a1](https://github.com/cockroachdb/cockroach/commits/7756496a7d7648c7cc19c83a184deb93817d30a1):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:1089
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:1052
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:1027
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1868
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).DescsTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1782
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.DescsTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:3902
        	            	  | github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster.(*Tenant).LookupTableDescriptorByID
        	            	  | 	github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster/tenant_state.go:197
        	            	  | pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test.TestDataDriven.func2.1
        	            	  | 	pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test/pkg/ccl/spanconfigccl/spanconfigsplitterccl/datadriven_test.go:126
        	            	  | github.com/cockroachdb/datadriven.RunTest.func1
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:118
        	            	  | github.com/cockroachdb/datadriven.runDirective.func1
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:337
        	            	  | github.com/cockroachdb/datadriven.runDirective
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:342
        	            	  | github.com/cockroachdb/datadriven.runDirectiveOrSubTest
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:211
        	            	  | github.com/cockroachdb/datadriven.runTestInternal
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:184
        	            	  | github.com/cockroachdb/datadriven.RunTestAny
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:144
        	            	  | github.com/cockroachdb/datadriven.RunTest
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:117
        	            	  | pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test.TestDataDriven.func2
        	            	  | 	pkg/ccl/spanconfigccl/spanconfigsplitterccl/spanconfigsplitterccl_test/pkg/ccl/spanconfigccl/spanconfigsplitterccl/datadriven_test.go:90
        	            	  | github.com/cockroachdb/datadriven.Walk.func1
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:425
        	            	  | github.com/cockroachdb/datadriven.WalkAny
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:436
        	            	  | github.com/cockroachdb/datadriven.WalkAny.func1
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:449
        	            	  | github.com/cockroachdb/datadriven.subTest.func1
        	            	  | 	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/test_interface.go:27
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (3) relation "[107]" does not exist
        	            	Error types: (1) *pgerror.withCandidateCode (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestDataDriven/tables
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/88cfb1c77b9d258ed6991015ff838b5f/logTestDataDriven_tables2470488008
    --- FAIL: TestDataDriven/tables (2.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38562

--- a/pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster/cluster.go
+++ b/pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster/cluster.go
@@ -61,10 +61,10 @@ func (h *Handle) InitializeTenant(ctx context.Context, tenID roachpb.TenantID) *
 		tenantState.cleanup = func() {} // noop
 	} else {
 		serverGCJobKnobs := testServer.SystemLayer().TestingKnobs().GCJob
-		tenantGCJobKnobs := sql.GCJobTestingKnobs{SkipWaitingForMVCCGC: true}
+		// Copy the GC job knobs from the server to the tenant.
+		tenantGCJobKnobs := sql.GCJobTestingKnobs{}
 		if serverGCJobKnobs != nil {
 			tenantGCJobKnobs = *serverGCJobKnobs.(*sql.GCJobTestingKnobs)
-			tenantGCJobKnobs.SkipWaitingForMVCCGC = true
 		}
 		tenantArgs := base.TestTenantArgs{
 			TenantID: tenID,

